### ISSUE TITLE
Upgrade to 0.7.26 breaks d3-drag

### ISSUE THREAD

**Issue by mydea** [emberjs/ember-test-helpers#426]
After upgrading from 0.7.25, our tests started failing. It was really difficult to track down, but it seems to be related to changes in this addon.

Basically, we started getting a lot of errors like this: 

`Uncaught TypeError: Cannot read property \'document\' of null at http://localhost:7357/assets/vendor.js, line 78363'`

This seems to be tied to the `click` helper and https://github.com/d3/d3-drag.

Which somehow failed here: (it's hard to say exactly since this was always minified in our app, but we tried to trace the steps as good as possible): https://github.com/d3/d3-drag/blob/ec20d5e35bfc6ad2b4266494aef475c94cc2e6c3/src/nodrag.js#L16

Which is called here: https://github.com/d3/d3-drag/blob/ec20d5e35bfc6ad2b4266494aef475c94cc2e6c3/src/drag.js#L74  (or, I guess it could also have been this code: https://github.com/d3/d3-drag/blob/ec20d5e35bfc6ad2b4266494aef475c94cc2e6c3/src/drag.js#L55)

Basically, our test just did this:

```
await click('[data-test-d3-circle-element]');
```

Which lead it to blow up. It works if I do it manually, it seems the `click` listener has somehow changed to break d3. This used to work fine in 0.7.25.

**Comment by mydea** [comment]
I made a small repo to reproduce this:
https://github.com/mydea/ember-test-helpers-click-reproduction

Note that there is one failing test. It works manually. You can get the test to pass by adding this to the package.json:

```
"resolutions": {
    "**/@ember/test-helpers": "0.7.25"
  }
```

The code is in `app/components/d3-drag-test.js`:

```js
import { select, mouse } from 'd3-selection';
import { drag } from 'd3-drag'

let selection = select(this.element);

selection.call(
  drag()
    .on('start', () => this.onDragStart())
    .on('drag', () => {
      let [cx, cy] = mouse(this.element);
      this.onDrag(cx, cy);
    })
    .on('end', () => this.onDragEnd())
);
```

And the corresponding test that breaks in 0.7.26 is just:

```js
 test('it renders', async function(assert) {
    assert.expect(0);
    await render(hbs`{{d3-drag-test}}`);
    await click('circle');
  });
```

**Comment by rwjblue** [comment]
Hmm. This is a tad tricky to quickly debug since there are so many changes included between 0.7.25 and 0.7.26.

My _guess_ (from the symptoms reported) is that its related to #387 somehow. I'll try to clone the reproduction and step through...

**Comment by rwjblue** [comment]
Just confirmed via the reproduction that this is related to #387. After #387 we are firing "real" `MouseEvent`'s, but we do not initialize the `view` property (the DOM itself [defaults it to `null`](https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/UIEvent).

Changing the test to the following passes:

```js
  test('it renders', async function(assert) {
    assert.expect(0);
    await render(hbs`{{d3-drag-test}}`);
    await click('circle', { view: window });
  });
```

Details:

* https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/MouseEvent
* https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/UIEvent

---

Should we pass `view` by default for `MouseEvent`'s?

/ cc @ggayowsky / @cibernox

**Comment by ggayowsky** [comment]
Looking at the MDN documents, they suggest passing the `view` parameter for a simulated MouseEvent, so I think that is the correct option.

https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent#Example

I am guessing the fix would probably be changing the `eventOpts` assignment in `buildMouseEvent` from
```
let eventOpts = assign({}, DEFAULT_EVENT_OPTIONS, options);
```
to
```
let eventOpts = assign({ view: window }, DEFAULT_EVENT_OPTIONS, options);
```

I would make the change, however, don't have the repo checked out or the time for the next few weeks.

**Comment by rwjblue** [comment]
@ember/test-helpers 0.7.27 published 🎉
